### PR TITLE
Demo Config : Adding AD Indices to system index and creating pre-defined roles

### PR DESCRIPTION
**Pull request by saratvemulapalli**
*Issue https://github.com/opendistro-for-elasticsearch/anomaly-detection/issues/195*

*Description of changes:*
Moving AD indices to system index and defining pre-defined roles for Anomaly Detection plugin users.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by sujithvm — DISMISSED** [review]

File: tools/install_demo_configuration.sh [hunk 1]
```diff
@@ -373,7 +373,7 @@ echo "opendistro_security.enable_snapshot_restore_privilege: true" | $SUDO_CMD t
 echo "opendistro_security.check_snapshot_restore_write_privileges: true" | $SUDO_CMD tee -a "$ES_CONF_FILE" > /dev/null
 echo 'opendistro_security.restapi.roles_enabled: ["all_access", "security_rest_api_access"]' | $SUDO_CMD tee -a "$ES_CONF_FILE" > /dev/null
 echo 'opendistro_security.system_indices.enabled: true' | $SUDO_CMD tee -a "$ES_CONF_FILE" > /dev/null
-echo 'opendistro_security.system_indices.indices: [".opendistro-alerting-config", ".opendistro-alerting-alert*"]' | $SUDO_CMD tee -a "$ES_CONF_FILE" > /dev/null
+echo 'opendistro_security.system_indices.indices: [".opendistro-alerting-config", ".opendistro-alerting-alert*", ".opendistro-anomaly-results", ".opendistro-anomaly-detector*"]' | $SUDO_CMD tee -a "$ES_CONF_FILE" > /dev/null
```
Comment (ylwu-amzn) [inline tools/install_demo_configuration.sh:376]: `.opendistro-anomaly-results` -> `.opendistro-anomaly-results*` ?
`.opendistro-anomaly-detector*` -> `.opendistro-anomaly-detector` ?

Comment (saratvemulapalli) [reply tools/install_demo_configuration.sh:376]: Sure for results its makes sense as we have other history indices. 
for `.opendistro-anomaly-detector*` it includes both  `.opendistro-anomaly-detector-jobs` and `.opendistro-anomaly-detectors`

**Review by vrozov — DISMISSED** [review]

File: tools/install_demo_configuration.sh (hunk 1 above)
Comment (ylwu-amzn) [reply tools/install_demo_configuration.sh]: Got it, make sense.

**Review by sujithvm — APPROVED** [review]

File: securityconfig/roles.yml [hunk 2]
```diff
@@ -38,3 +38,25 @@ alerting_full_access:
         - 'indices_monitor'
         - 'indices:admin/aliases/get'
         - 'indices:admin/mappings/get'
+
+# Allow users to read Anomaly Detection detectors and results
+ad_read_access:
```
Comment (vrozov) [inline securityconfig/roles.yml:43]: `ad` -> `anomaly`

Comment (saratvemulapalli) [reply securityconfig/roles.yml:43]: yeah we internally use ad as anomaly detection, do you recommend expanding it?

Comment (vrozov) [reply securityconfig/roles.yml:43]: yes, 'ad' is too short and may mean not only anomaly detection.

Comment (saratvemulapalli) [reply securityconfig/roles.yml:43]: Sure will rename it do anomaly.

**Comment by saratvemulapalli** [comment]
Thanks guys for taking a look. 
Addressed all comments.

**Review by ylwu-amzn — APPROVED** [review]

**Review by vrozov — APPROVED** [review]

**Pull request merged**